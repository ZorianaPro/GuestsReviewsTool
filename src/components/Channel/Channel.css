.Channel {
  position: relative;
  display: flex;
}

.Channel .AirbnbSVG {
  width: 5em;
  height: 2.5em;
}

.Channel .BookingcomSVG {
  width: 6em;
  height: 2.5em;
}

.Channel .HoliduSVG {
  width: 5em;
  height: 2.5em;
}

