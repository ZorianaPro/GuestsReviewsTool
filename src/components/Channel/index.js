import React from 'react';
import './Channel.css';
import AirbnbSVG from '../AirbnbSVG';
import BookingcomSVG from '../BookingcomSVG';
import HoliduSVG from '../HoliduSVG';

const Channel = ({
  channel
}) => (
  <div className="Channel">
    {
      channel === 'AIRBNB'
      && <AirbnbSVG/>
    }
    {
      channel === 'BOOKINGCOM'
      && <BookingcomSVG/>
    }
    {
      channel === 'HOLIDU'
      && <HoliduSVG/>
    }
  </div>
);

export default Channel;