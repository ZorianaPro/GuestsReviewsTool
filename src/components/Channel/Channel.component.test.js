import React from 'react';
import { shallow, mount } from 'enzyme';
import Channel from './';

describe('Channel', () => {
  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <Channel />)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(shallow(<Channel />)).toMatchSnapshot();
  });

  describe('channel is AIRBNB', () => {
    let channel;
    beforeEach(() => {
      channel = mount(<Channel channel="AIRBNB"/>);
    });

    it('renders AIRBNB icon', () => {
      expect(channel.exists('.AirbnbSVG'))
        .toBe(true);
    });
  });

  describe('channel is BOOKINGCOM', () => {
    let channel;
    beforeEach(() => {
      channel = mount(<Channel channel="BOOKINGCOM"/>);
    });

    it('renders BOOKINGCOM icon', () => {
      expect(channel.exists('.BookingcomSVG'))
        .toBe(true);
    });
  });
  describe('channel is HOLIDU', () => {
    let channel;
    beforeEach(() => {
      channel = mount(<Channel channel="HOLIDU"/>);
    });

    it('renders HOLIDU icon', () => {
      expect(channel.exists('.HoliduSVG'))
        .toBe(true);
    });
  });
});
