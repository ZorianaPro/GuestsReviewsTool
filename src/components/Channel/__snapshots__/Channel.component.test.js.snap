// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel renders as expected 1`] = `
<div
  className="Channel"
/>
`;
